pkgs/applications/science/logic/key/default.nix

```diff
@@ -0,0 +1,74 @@
+{ stdenv
+, fetchurl
+, unzip
+, jdk
+, ant
+, jre
+, makeWrapper
+, runCommand
+, key
```

Comment:
Not sure if that's a good idea

Reply:
The tricky bit is if you'd `pkgs.key.override x` and use it directly without inserting it into `pkgs` by means of an overlay, you'd have to do something like `let key = pkgs.key.override (x // { inherit key; }; in key`.

Instead, you could do

```
let gitRevision = # ...
    key = stdenv.mkDerivation rec {
      # ...
      passthru.tests.check-version = /* ... */;  # syntactically unchanged but uses the local binding
    }
in key
```

Reply:
don't we basically have the same problem with all nixosTests?

Reply:
That's an excuse rather than argument, but I'll leave it up to you.